components/equihash/src/tromp.rs

```diff
@@ -175,10 +175,16 @@ pub fn solve_200_9_compressed<const N: usize>(
     const DIGIT_BITS: usize = 200 / (9 + 1);
     let solutions = solve_200_9(input, next_nonce);
 
-    solutions
+    let mut solutions: Vec<Vec<u8>> = solutions
         .iter()
         .map(|solution| get_minimal_from_indices(solution, DIGIT_BITS))
-        .collect()
+        .collect();
+
+    // Just in case the solver returns solutions that become the same when compressed.
```

Comment:
I don't think this should ever be the case. The compressed representation should not be lossy, it just packs the indices together to remove the effective padding bits that exist in the `u32` representation.

Reply:
This was added in the same [commit](https://github.com/zcash/librustzcash/pull/1088/commits/d2c4765d16f20d86f87431920baa6f1d2ef516c4) as the `.dedup()` call on the list of uncompressed solutions, if you're confident that the C code only removes effective padding, it may not have been observed, or there could be a subtle difference between the C code and the Rust code in `compress_array()`.